### ISSUE TITLE
Letters

### ISSUE THREAD

**Issue by loopier** [loopier/animatron-godot3#20]
I'm working on some frames of animated letters.  I'm expecting to do the whole ascii code.  It's hand-drawn, nothing fancy, we could think of a command to write with these, maybe `/write word`.  What would be the best way to implement it?  This would not be used to type with fonts, but with sprite sheets or sequences of images.  Each sequence is just one letter, so it gives a shaky effect.  Things that come to mind:

-  Many letters would need to be put on stage one after the other, but positioning is not going to be very accurate because it's not a font, just a bunch of square images.
- Spaces should be taken into account.
- Each letter could have its own FPS.
- Each letter is an actor, actually.
- Should spacing be controllable?  
- Should spacing be related to image width?
- Having static text (no  shaky effect) with all the letters standing still but chosen randomly when created.  This way, two instances of the same letter would be a little different and give the impression to be handwritten.
- The letters use could actually be any sequence or sprite sheet.  Maybe the animations to be used could be defined with a `def`?  
- The previous point could be a fast way to create an array of images quickly, mapping each character to an image sequence, usually a letter, but not restricted to that.

Any thoughts?

**Comment by totalgee** [comment]
Not the same thing, but sort of related: I also wanted something with letters/text, in this case probably drawing from fonts, for various reasons. One of them would be to draw the code (commands) being executed. Not the same as your idea, but just a reminder we may have *various* text needs.

**Comment by totalgee** [comment]
For yours, I think maybe we should have a special kind of asset that is the whole alphabet (possibly with animated letters, as you suggest). Then code that knows how to use them, deal with simple layout and spaces, etc.

**Comment by loopier** [comment]
> Not the same thing, but sort of related: I also wanted something with letters/text, in this case probably drawing from fonts, for various reasons. One of them would be to draw the code (commands) being executed. Not the same as your idea, but just a reminder we may have _various_ text needs.

Yes, we need that, too.  We listed it in #6.  We can kind of  do that already with `/say`.  Not the best tool for it, but the base is there.



> For yours, I think maybe we should have a special kind of asset that is the whole alphabet (possibly with animated letters, as you suggest). Then code that knows how to use them, deal with simple layout and spaces, etc.

I'd prefer a special command over a special asset.  If all assets are of just  one type, we can use any commands with them, which opens up the creative possibilities.  Maybe the command should have a list of assets and work on those?  Maybe a `def`-ish map to map ascii chars to actors?

```
def /alphabet/a actor-or-asset
    /alphabet/b actor-or-asset
   ...
    /alphabet/A actor-or-asset
   ...
    /alphabet/( actor-or-asset // or /alphabet/openningparenthesis
```
